### PR TITLE
remove egress sds

### PR DESCRIPTION
**Pull request by richardwxn**
we not yet support egress sds, right? need confirmation from @incfly @pitlv2109

### CODE REVIEW

File: gateways/istio-egress/templates/deployment.yaml [hunk 1]
```diff
@@ -197,12 +197,6 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.namespace
-          {{- if $gateway.sds }}
-          {{- if $gateway.sds.enabled }}
```
Comment (incfly) [inline gateways/istio-egress/templates/deployment.yaml]: `gateway.sds.enabled` used for public facing certificate for ingress. From the doc I can tell, https://istio.io/docs/tasks/traffic-management/ingress/secure-ingress-sds/

We do support SDS, for egress gateway, for workload identity, between workload and egress, but that's governed already by `if .Values.global.sds.enabled`  below.

I can't think of a use case to support https cert on egress, seems abnormal. 

@JimmyCYJ WDTY

**Comment by incfly** [comment]
Does this cause any problem/failures, If just clean up, then might not worth the risk?

**Comment by richardwxn** [comment]
@incfly It is a clean up because it doesn't seem to be used

**Comment by pitlv2109** [comment]
I don't think we have sds for egress (to provision and exchange certs with external services), but @JimmyCYJ can confirm.

File: gateways/istio-egress/values.yaml [hunk 2]
```diff
@@ -56,20 +56,6 @@ gateways:
       secretName: istio-egressgateway-ca-certs
       mountPath: /etc/istio/egressgateway-ca-certs
 
-    sds:
-      # If true, ingress gateway fetches credentials from SDS server to handle TLS connections.
-      enabled: false
-      # SDS server that watches kubernetes secrets and provisions credentials to ingress gateway.
```
Comment (pitlv2109) [inline gateways/istio-egress/values.yaml]: Maybe this is a copy paste error from the ingress, as you can see all the comments are for ingress

File: gateways/istio-egress/templates/deployment.yaml (hunk 1 above)
Comment (JimmyCYJ) [reply gateways/istio-egress/templates/deployment.yaml]: @incfly We don't support SDS for external traffic at egress. There is a tracking GH issue https://github.com/istio/istio/issues/14039.

**Review by incfly — APPROVED** [review]

File: gateways/istio-egress/templates/deployment.yaml (hunk 1 above)
Comment (incfly) [reply gateways/istio-egress/templates/deployment.yaml]: Thanks confirmation. Approving.

**Review by incfly — APPROVED** [review]

**Comment by richardwxn** [comment]
looks like the helm lint is failing for no entry for map for new PRs, I think this was fixed long time ago in helm, might be a helm lint regression?

**Review by morvencao — APPROVED** [review]
Approve this since all comments has been addressed.

**Pull request merged**